### PR TITLE
Feature/IAC-693 mvn vra ng pull do not work for project entitlements for new content sharing policy in vra 8.8.2

### PR DESCRIPTION
**Pull request by gowthamr-vmware**
### Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added relevant error handling and logging messages to help troubleshooting
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have updated CHANGELOG.md with a short summary of the changes introduced
- [x] I have tested against live environment, if applicable
- [x] I have added relevant usage information (As-built)
- [x] Any structure and/or content vRA-NG improvements are synchronized with vra-ng and ts-vra-ng archetypes
- [x] Every new or updated Installer property is documented in docs/archive/doc/markdown/use-bundle-installer.md
- [x] Dependencies in pom.xml are up-to-date
- [x] My changes have been rebased and squashed to the minimal number of relevant commits
- [x] My changes have a descriptive commit message with a short title, including a `Fixed #XXX -` or `Closed #XXX -` prefix to auto-close the issue

### CODE REVIEW

File: docs/versions/latest/Release.md [hunk 1]
```diff
@@ -39,7 +48,47 @@
 [//]: # (Optional But higlhy recommended Specify *NONE* if missing)
 [//]: # (#### Relevant Documentation:)
 
+### Enabled Unit testing for npm lib based projects
```
Comment (Michaelpalacce) [inline docs/versions/latest/Release.md:51]: These release notes are from the previous release. Perhaps an issue with the sync?

Comment (gowthamr-vmware) [reply docs/versions/latest/Release.md]: Hi Stefan. Yeah, this added when I synched the latest changes from main. I have changed it now. Thanks!

File: common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/model/vrang/VraNgPackageDescriptor.java [hunk 2]
```diff
@@ -118,6 +119,10 @@ public void setCatalogEntitlement(List<String> catalogEntitlement) {
         this.catalogEntitlement = catalogEntitlement;
     }
 
+	public VraNgPolicy getPolicy() {
+        return this.policy;
+    }
+
```
Comment (ivo-kotev) [inline common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/model/vrang/VraNgPackageDescriptor.java:125]: Following the existing structure, can we have both getter and setter?

File: maven/archetypes/vra-ng/src/main/resources/archetype-resources/content.yaml [hunk 3]
```diff
@@ -39,3 +43,5 @@ resource-action:
 catalog-entitlement:
 content-source:
 property-group:
+policy:
+  content-sharing:
```
Comment (ivo-kotev) [inline maven/archetypes/vra-ng/src/main/resources/archetype-resources/content.yaml:47]: Can you please add the same here: https://github.com/vmware/build-tools-for-vmware-aria/blob/main/maven/archetypes/ts-vra-ng/src/main/resources/archetype-resources/vra/content.yaml

**Comment by ivo-kotev** [comment]
Great content. Is it possible to add in the unit test the content of the API response for the content sharing policy, and then to validate the expected content that we are going to store in Git once the content is parsed?

Generally we should be removing any content that can be expected incompatible between systems - like Usernames, user groups, etc. (even the policy ID - in case during import it's not taken into account by Aria Automation

**Review by ivo-kotev — CHANGES_REQUESTED** [review]
Great content. Is it possible to add in the unit test the content of the API response for the content sharing policy, and then to validate the expected content that we are going to store in Git once the content is parsed?

Generally we should be removing any content that can be expected incompatible between systems - like Usernames, user groups, etc. (even the policy ID - in case during import it's not taken into account by Aria Automation

File: common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/model/vrang/VraNgPackageDescriptor.java (hunk 2 above)
Comment (gowthamr-vmware) [reply common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/model/vrang/VraNgPackageDescriptor.java:125]: Hi Ivo, I have added a test case for RestClientVraNgprimitive to compare the content of API response. I have already removed few properties which are incompatible. Please let me know, if you need any other properties that needs to be removed. Also, I have added the setter property for setpolicy().

File: maven/archetypes/vra-ng/src/main/resources/archetype-resources/content.yaml (hunk 3 above)
Comment (gowthamr-vmware) [reply maven/archetypes/vra-ng/src/main/resources/archetype-resources/content.yaml:47]: I have made the changes in the file you specified above.

**Comment by gowthamr-vmware** [comment]
@ivo-kotev : I have made the required changes and tested it using push/pull/installer options. Please check and let me know if anything needs to be done further.

**Review by ivo-kotev — CHANGES_REQUESTED** [review]
I was testing to push the content source to a clean environment.

The content is created successfully, but it's not validating the ID that must be used when new content source is created. As a result, the sharing policy is created, but the link to the content source is with a wrong ID.

Please update the logic, so before pushing the sharing policy, it's payload is updated with the correct content source id.

**Comment by gowthamr-vmware** [comment]
@ivo-kotev : Do you mean to say that, I need to add logic to check if the payload's content source ID is present in the system before pushing the content sharing policy?

**Comment by ivo-kotev** [comment]
> @ivo-kotev : Do you mean to say that, I need to add logic to check if the payload's content source ID is present in the system before pushing the content sharing policy?
> 
> Let me know if my understanding is right?

Yes, you need to check if the ID is present - if yes - no action, if the id is missing, you need to check if in the current package we have a content source defined - grab it's name and find the proper ID for it - then use this id in the policy

**Comment by Michaelpalacce** [comment]
I am not sure how to test this, but I can't pull anything right now from vRA. @gowthamr-vmware has this been tested? How?

**Pull request merged**